scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,217 @@
+            or line.find('LF') >= 0)
+
+
+def get_test_file_name(test_path):
+    """Get the file name from the absolute path.
+
+    Args:
+        test_path: string. The file's absolute path.
+
+    Returns:
+        String. The file path.
+    """
+    if not test_path:
+        sys.stderr.write(
+            'The test path is empty or null.'
+            'It\'s not possible to diff the test coverage correctly.')
+        sys.exit(1)
+
+    return test_path[test_path.rindex('/') + 1:]
+
+
+def get_lcov_file_tests(file_path):
+    """Get all tests from a lcov file, and filters it to return only:
+    - File path
+    - File total lines
+    - File covered lines
+
+    Args:
+      file_path: string. The path of lcov file.
+
+    Returns:
+      String array. An array with all tests filtered, including only important
+      data for the diff.
+    """
+    with python_utils.open_file(file_path, 'r') as f:
+        tests_array = f.read().split('end_of_record')
+        tests_array_filtered = []
+
+        for test in tests_array:
+            lines = [line for line in test.splitlines() if filter_lines(line)]
+            if len(lines) > 0:
+                tests_array_filtered.append(lines)
+
+        return tests_array_filtered
+
+
+def build_tests_fully_coverage_dict():
+    """Build a dict with only fully covered files from develop branch.
+
+    Returns:
+        Dictionary. A dict containing file path, total lines and covered lines
+        of each tested file.
+
+    Raises:
+      Exception: If DEV_LCOV_FILE_PATH doesn't exist.
+    """
+    coverage_dict = {}
+
+    if not os.path.exists(DEV_LCOV_FILE_PATH):
+        raise Exception(
+            'File at path {} doesn\'t exist'.format(DEV_LCOV_FILE_PATH))
+
+    tests = get_lcov_file_tests(DEV_LCOV_FILE_PATH)
+    for lines in tests:
+        total_lines = lines[1].split(':')[1]
+        covered_lines = lines[2].split(':')[1]
+
+        if total_lines == covered_lines:
+            test_name = get_test_file_name(lines[0])
+            coverage_dict[test_name] = [
+                int(total_lines),
+                int(covered_lines)
+            ]
+
+    return coverage_dict        
+
+
+def lcov_files_diff():
+    """Check if any 100% covered file had the coverage dropped.
+
+    Raises:
+      Exception: If PR_LCOV_FILE_PATH doesn't exist.
+    """
+    fully_covered_tests = build_tests_fully_coverage_dict()
+
+    if not os.path.exists(PR_LCOV_FILE_PATH):
+        raise Exception('File at path {} doesn\'t exist'.format(
+            PR_LCOV_FILE_PATH))
+
+    tests = get_lcov_file_tests(PR_LCOV_FILE_PATH)
+    for lines in tests:
+        test_name = get_test_file_name(lines[0])
+
+        if test_name in fully_covered_tests:
+            total_lines = lines[1].split(':')[1]
+            covered_lines = lines[2].split(':')[1]
+
+            test = fully_covered_tests[test_name]
+            total_lines_test = test[0]
+
+            if (total_lines_test == int(total_lines)
```

Comment:
This isn't enough right? The number of lines in the test might increase, as a result of the change in the file, but the coverage could drop. Also, in case of deletions, the number of lines would reduce and again the coverage would drop. A better check is that if test_name is in fully_covered_tests, then int(total_lines) must equal int(covered_lines).

Reply:
Done